lib/api-sdk/projectTeamHooks.tsx

```diff
@@ -0,0 +1,94 @@
+import useSWR, { mutate } from "swr";
+import { Routes, APIGet, APIPost, APIPut, APIDelete } from "./fetcher";
+import { NextApiRequest } from "next";
+import { useUpdateEffect } from "react-use";
+
+type SelfHookParams<T> = {
+  defaultOnError?: (errorMsg: string) => void;
+  initialModel?: T;
+  req?: NextApiRequest;
+};
+
+function useProjectTeamSelf({
+  defaultOnError,
+  initialModel
+}: SelfHookParams<ProjectTeam>) {
+  const resourceRoute = Routes.ProjectTeamSelf;
+
+  const { data: projectTeam, error } = useSWR<ProjectTeam, any>(
+    resourceRoute,
+    async () => {
+      const res = await APIGet<ProjectTeam>(resourceRoute);
+      if (res.success) {
+        return res.success;
+      }
+      throw res.error;
+    },
+    {
+      initialData: initialModel
+    }
+  );
+
+  useUpdateEffect(() => {
+    if (error && error != "") defaultOnError(error), [error];
+  });
+
+  // LOOK AT THIS HIGHER ORDER FUNCTION
+  function fetchWithMutation<T extends (...args: any[]) => any>(
```

Comment:
so just to double check my clarification...

if using a route that does mutation, this will hit that route, update the data layer via `mutate`, therefore making sure any components relying on that state in the data layer update -- right?

Reply:
mutate tells useSWR that the resource has been updated, then since we have semantics that PUT/POST return the updated resource, we invoke mutate with the new value. This then triggers within useSWR basically setData(newValue). So the data is never stale